lib/src/utils.dart

```diff
@@ -90,17 +90,22 @@ String winstonToAr(BigInt winston) {
 
 /// Safely get the error from an Arweave HTTP response.
 String getResponseError(Response res) {
-  if (res.headers['Content-Type'] == 'application/json') {
-    Map<String, dynamic> errJson = json.decode(res.body);
-
-    if (errJson['data'] != null) {
-      return errJson['data'] is Map
-          ? errJson['data']['error']
-          : errJson['data'];
+  try {
+    if (res.headers['Content-Type'] == 'application/json') {
+      Map<String, dynamic> errJson = json.decode(res.body);
+
+      if (errJson['data'] != null) {
+        return errJson['data'] is Map
+            ? errJson['data']['error']
+            : errJson['data'];
+      }
     }
-  }
 
-  return res.body;
+    return res.body;
+  } catch (e) {
+    print('Unknown upload error ${e.toString()}.');
+    return e.toString();
```

Comment:
Note: This could break an implicit contract for the format of the data returned by this function based on the `res.headers['Content-Type']` setting. I think it was a weird choice for this function to originally fall back to returning a response body if the Content-Type isn't `application/json`. Perhaps some validation should be happening on res.body to make sure that it doesn't throw if there's no data in it? e.g. just return an empty string?

Reply:
[Removed the changes here since as you mentioned, this is broken](https://github.com/ardriveapp/arweave-dart/pull/14/commits/14e732fbc05fd89c904b76b4ca04ac5abebdb288)